Issue trying to create package with windows7+msys2

Hi, after trying to follow carefully the instructions at https://github.com/keepassxreboot/keepassxc/wiki/Set-up-Build-Environment-on-Windows and installing all the corresponding packages (as well as wix) I've managed to build keepasxc as well as many other targets +/- ok.

Now, I haven't been able to manage to make a package out of it, please take a look to the log when I try to build `keepasxc-proxy`:

```
$ cmake.exe --build . --target keepassxc-proxy -- -j8 VERBOSE=1
/D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -H/C/msys64/home/KneDa/keepassxc -B/C/msys64/home/KneDa/keepassxc/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/make -f CMakeFiles/Makefile2 keepassxc-proxy
make[1]: Entering directory '/home/KneDa/keepassxc/build'
/D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -H/C/msys64/home/KneDa/keepassxc -B/C/msys64/home/KneDa/keepassxc/build --check-build-system CMakeFiles/Makefile.cmake 0
/D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -E cmake_progress_start /C/msys64/home/KneDa/keepassxc/build/CMakeFiles 3
/usr/bin/make -f CMakeFiles/Makefile2 src/proxy/CMakeFiles/keepassxc-proxy.dir/all
make[2]: Entering directory '/home/KneDa/keepassxc/build'
/usr/bin/make -f src/proxy/CMakeFiles/proxy_autogen.dir/build.make src/proxy/CMakeFiles/proxy_autogen.dir/depend
make[3]: Entering directory '/home/KneDa/keepassxc/build'
/D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -E cmake_depends "MSYS Makefiles" /C/msys64/home/KneDa/keepassxc /C/msys64/home/KneDa/keepassxc/src/proxy /C/msys64/home/KneDa/keepassxc/build /C/msys64/home/KneDa/keepassxc/build/src/proxy /C/msys64/home/KneDa/keepassxc/build/src/proxy/CMakeFiles/proxy_autogen.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/home/KneDa/keepassxc/build'
/usr/bin/make -f src/proxy/CMakeFiles/proxy_autogen.dir/build.make src/proxy/CMakeFiles/proxy_autogen.dir/build
make[3]: Entering directory '/home/KneDa/keepassxc/build'
[ 33%] Automatic MOC and UIC for target proxy
cd /C/msys64/home/KneDa/keepassxc/build/src/proxy && /D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -E cmake_autogen C:/msys64/home/KneDa/keepassxc/build/src/proxy/CMakeFiles/proxy_autogen.dir/AutogenInfo.cmake Release
make[3]: Leaving directory '/home/KneDa/keepassxc/build'
[ 33%] Built target proxy_autogen
/usr/bin/make -f src/proxy/CMakeFiles/proxy.dir/build.make src/proxy/CMakeFiles/proxy.dir/depend
make[3]: Entering directory '/home/KneDa/keepassxc/build'
/D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -E cmake_depends "MSYS Makefiles" /C/msys64/home/KneDa/keepassxc /C/msys64/home/KneDa/keepassxc/src/proxy /C/msys64/home/KneDa/keepassxc/build /C/msys64/home/KneDa/keepassxc/build/src/proxy /C/msys64/home/KneDa/keepassxc/build/src/proxy/CMakeFiles/proxy.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/home/KneDa/keepassxc/build'
/usr/bin/make -f src/proxy/CMakeFiles/proxy.dir/build.make src/proxy/CMakeFiles/proxy.dir/build
make[3]: Entering directory '/home/KneDa/keepassxc/build'
make[3]: Nothing to be done for 'src/proxy/CMakeFiles/proxy.dir/build'.
make[3]: Leaving directory '/home/KneDa/keepassxc/build'
[ 66%] Built target proxy
/usr/bin/make -f src/proxy/CMakeFiles/keepassxc-proxy_autogen.dir/build.make src/proxy/CMakeFiles/keepassxc-proxy_autogen.dir/depend
make[3]: Entering directory '/home/KneDa/keepassxc/build'
/D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -E cmake_depends "MSYS Makefiles" /C/msys64/home/KneDa/keepassxc /C/msys64/home/KneDa/keepassxc/src/proxy /C/msys64/home/KneDa/keepassxc/build /C/msys64/home/KneDa/keepassxc/build/src/proxy /C/msys64/home/KneDa/keepassxc/build/src/proxy/CMakeFiles/keepassxc-proxy_autogen.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/home/KneDa/keepassxc/build'
/usr/bin/make -f src/proxy/CMakeFiles/keepassxc-proxy_autogen.dir/build.make src/proxy/CMakeFiles/keepassxc-proxy_autogen.dir/build
make[3]: Entering directory '/home/KneDa/keepassxc/build'
[ 66%] Automatic MOC and UIC for target keepassxc-proxy
cd /C/msys64/home/KneDa/keepassxc/build/src/proxy && /D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -E cmake_autogen C:/msys64/home/KneDa/keepassxc/build/src/proxy/CMakeFiles/keepassxc-proxy_autogen.dir/AutogenInfo.cmake Release
make[3]: Leaving directory '/home/KneDa/keepassxc/build'
[ 66%] Built target keepassxc-proxy_autogen
/usr/bin/make -f src/proxy/CMakeFiles/keepassxc-proxy.dir/build.make src/proxy/CMakeFiles/keepassxc-proxy.dir/depend
make[3]: Entering directory '/home/KneDa/keepassxc/build'
/D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -E cmake_depends "MSYS Makefiles" /C/msys64/home/KneDa/keepassxc /C/msys64/home/KneDa/keepassxc/src/proxy /C/msys64/home/KneDa/keepassxc/build /C/msys64/home/KneDa/keepassxc/build/src/proxy /C/msys64/home/KneDa/keepassxc/build/src/proxy/CMakeFiles/keepassxc-proxy.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/home/KneDa/keepassxc/build'
/usr/bin/make -f src/proxy/CMakeFiles/keepassxc-proxy.dir/build.make src/proxy/CMakeFiles/keepassxc-proxy.dir/build
make[3]: Entering directory '/home/KneDa/keepassxc/build'
[100%] Linking CXX executable keepassxc-proxy.exe
cd /C/msys64/home/KneDa/keepassxc/build/src/proxy && /D/software/vcpkg/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe -E remove -f CMakeFiles/keepassxc-proxy.dir/objects.a
cd /C/msys64/home/KneDa/keepassxc/build/src/proxy && /C/msys64/mingw64/bin/ar.exe cr CMakeFiles/keepassxc-proxy.dir/objects.a @CMakeFiles/keepassxc-proxy.dir/objects1.rsp
cd /C/msys64/home/KneDa/keepassxc/build/src/proxy && /C/msys64/mingw64/bin/g++.exe  -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -std=c++11 -fsized-deallocation -Wno-deprecated-declarations -O3 -DNDEBUG   -Wl,--nxcompat -Wl,--dynamicbase -Wl,--high-entropy-va -Wl,--whole-archive CMakeFiles/keepassxc-proxy.dir/objects.a -Wl,--no-whole-archive  -o keepassxc-proxy.exe -Wl,--out-implib,libkeepassxc-proxy.dll.a -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles/keepassxc-proxy.dir/linklibs.rsp
collect2.exe: error: ld returned 5 exit status
make[3]: *** [src/proxy/CMakeFiles/keepassxc-proxy.dir/build.make:108: src/proxy/keepassxc-proxy.exe] Error 1
make[3]: *** Deleting file 'src/proxy/keepassxc-proxy.exe'
make[3]: Leaving directory '/home/KneDa/keepassxc/build'
make[2]: *** [CMakeFiles/Makefile2:605: src/proxy/CMakeFiles/keepassxc-proxy.dir/all] Error 2
make[2]: Leaving directory '/home/KneDa/keepassxc/build'
make[1]: *** [CMakeFiles/Makefile2:617: src/proxy/CMakeFiles/keepassxc-proxy.dir/rule] Error 2
make[1]: Leaving directory '/home/KneDa/keepassxc/build'
make: *** [Makefile:353: keepassxc-proxy] Error 2
```

Therefore when trying to do `make package` it'll stop at that point as well `[ 58%] Linking CXX executable keepassxc-proxy.exe`

Could you please advice?

Ns. I've also experienced some issues (compiler gets stuck) when trying to use ccache so right now I've just disabled, is this one familiar?

@droidmonkey Ok, in addition to my previous comment.

- Cleaned the build directory (ie: rm -rf * as well as hidden content)
- `$ cmake -G"Unix Makefiles" -DWITH_XC_ALL=ON -DCMAKE_BUILD_TYPE=Release ..
`
- Set CCACHE_FOUND to emtpy
- `make j8 package`

Good news is I don't have more linker errors... bad ones is I'm facing a similar error I was facing few hours ago:

```
Run CPack packaging tool...
CPack: Create package using ZIP
CPack: Install projects
CPack: - Run preinstall target for: KeePassXC
CPack: - Install project: KeePassXC []
Unable to find the platform plugin.
CMake Error at C:/msys64/home/KneDa/keepassxc/build/src/cmake_install.cmake:63 (file):
  file INSTALL cannot find
  "C:/msys64/home/KneDa/keepassxc/src/OPENSSL_DLL-NOTFOUND": No error.
Call Stack (most recent call first):
  C:/msys64/home/KneDa/keepassxc/build/cmake_install.cmake:37 (include)


CPack Error: Error when generating package: KeePassXC
make: *** [Makefile:158: package] Error 1

```

Could you please advice? Thx

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -487,8 +487,16 @@ if(MINGW)
     install(FILES ${Qt5_PREFIX}/ssl/certs/ca-bundle.crt DESTINATION "ssl/certs")
     # install OpenSSL library
     if(WITH_XC_NETWORKING)
-        find_library(OPENSSL_DLL NAMES libssl-1_1.dll libssl-1_1-x64.dll PATH_SUFFIXES bin)
-        find_library(CRYPTO_DLL NAMES libcrypto-1_1.dll libcrypto-1_1-x64.dll PATH_SUFFIXES bin)
+        find_file(OPENSSL_DLL NAMES libssl-1_1.dll libssl-1_1-x64.dll)
+        if (NOT OPENSSL_DLL)
+            message(FATAL_ERROR "Cannot find libssl dll, ensure openssl is properly installed.")
+        endif()
+
+        find_file(CRYPTO_DLL NAMES libcrypto-1_1.dll libcrypto-1_1-x64.dll)
+        if (NOT OPENSSL_DLL)
+            message(FATAL_ERROR "Cannot find libcrypto dll, ensure libgcrypt is properly installed.")
+        endif()
+
         install(FILES ${OPENSSL_DLL} ${CRYPTO_DLL} DESTINATION ".")
     endif()
 endif()

